.github/workflows/run-tests-and-mypy-and-black.yml

```diff
@@ -1,10 +1,6 @@
 name: Run tests and mypy
 
-on:
-  push:
-    branches: [dev]
-  pull_request:
-    types: [opened, synchronize, reopened, ready_for_review]
+on: push
```

Comment:
Whatever you want tbh, just pointing it out in case you hadn't considered doing it.
And since I see we have occasionally several changes to old code in MR, presumably because the previous MR did not have all the linting and formatting applied

Reply:
I changed it (it took a while though, i'm becoming devops haha), however i cannot configure properly the mypy hook (https://github.com/pre-commit/mirrors-mypy) to match what is checking our current configuration.

Also I updated some dependencies for dev (flake8 especially), removed the flake8 import sort, replacing it by isort to be more consistent with the backend. Also i'm thinking about opening a ticket to switch from Mypy to Pyright, for consistency aswell. Lmk what do you think @awennersteen

Reply:
I went for MyPy originally because it's what mostly used outside the cloud team at Pasqal. But feel free to make it consistent with internal code!

You can now put devops engineer on your resume ;)